src/local/Matcher.ts

```diff
@@ -0,0 +1,64 @@
+interface Data<T = undefined> {
+  criteria: string | RegExp;
+  data: T;
+}
+
+interface Matcher {
+  add<T>(criteria: string, data?: T): void;
+  match<T = any>(criteria: string): T[];
+  remove<T>(criteria: string, data?: T): void;
+  isWildcardChannel(str: string): boolean;
+}
+
+class Matcher {
+  matchers: Data<any>[];
```

Comment:
Is this supposed to be public or private?

Reply:
It's supposed  to be private, but the test file tries access it so I removed private to get rid of errors.

Reply:
Perhaps it makes sense to keep the name with an underscore prefix in this case then, to communicate that it's not supposed to be used.

We could add a getter returning a frozen copy / readonly array, but I think it's a bit unnecessary in this case.